docs/model-dev-guide/debug-models.rst

```diff
@@ -4,37 +4,22 @@
  Debugging Models
 ##################
 
-Using Determined to debug models depends on your environment.
-
-Your code on a Determined cluster differs from typical training scripts in the following ways:
-
--  The code conforms to the Trial APIs as a subclass of the Determined ``Trial`` class, indirectly,
-   by using one of the concrete subclasses, such as
-   :class:`~determined.pytorch.deepspeed.DeepSpeedTrial`.
+Using Determined to train your model can introduce a number of failure points that aren't present
+when running training scripts locally. Your code on a Determined cluster differs from typical
+training scripts in the following ways:
 
 -  The code runs in a Docker container on another machine.
-
 -  Your model can run many times in a hyperparameter search.
-
 -  Your model can be distributed across multiple GPUs or machines.
 
-These debugging steps introduce code changes incrementally, working toward a fully functioning
-Determined model. Follow the nine steps as applicable to your environment:
-
--  Model-related Issues
-      -  `Step 1 - Verify that your code runs locally`_
-      -  `Step 2 - Verify that each Trial subclass method works locally`_
-      -  `Step 3 - Verify local test mode`_
-
--  Docker- or Cluster-related Issues
-      -  `Step 4 - Verify that the original code runs in a notebook or shell`_
-      -  `Step 5 - Verify that each Trial subclass method works in a notebook or shell`_
-      -  `Step 6 - Verify that local test mode works in a notebook or shell`_
+These debugging steps introduce environment and changes incrementally, working toward fully
+functioning distributed training on a Determined cluster. Follow the nine steps as applicable to
+your environment:
 
--  Higher-level Issues
-      -  `Step 7 - Verify that cluster test mode works with slots_per_trial set to 1`_
-      -  `Step 8 - Verify that a single-GPU experiment works`_
-      -  `Step 9 - Verify that a multi-GPU experiment works`_
+-  `Step 1 - Verify that your training script runs locally`_
+-  `Step 2 - Verify that your training script runs in a notebook or shell`_
+-  `Step 3 - Verify that you can submit a single-GPU experiment`_
+-  `Step 4 - Verify that you can submit a multi-GPU experiment`_
```

Comment:
wonder if it would make sense to link to the profiling doc in this page, as sort of a "performance debugging" step

Reply:
I think that could make sense.  I'm not going to make that change today though.